usr/lib/icsf_stdll/new_host.c

```diff
@@ -766,6 +766,35 @@ CK_RV SC_SetOperationState(STDLL_TokData_t *tokdata,
     return rc;
 }
 
+CK_RV SC_SessionCancel(STDLL_TokData_t *tokdata, ST_SESSION_HANDLE *sSession,
+                       CK_FLAGS flags)
+{
+    SESSION *sess = NULL;
+    CK_RV rc = CKR_OK;
+
+    if (tokdata->initialized == FALSE) {
+        TRACE_ERROR("%s\n", ock_err(ERR_CRYPTOKI_NOT_INITIALIZED));
+        rc = CKR_CRYPTOKI_NOT_INITIALIZED;
+        goto done;
+    }
+
+    sess = session_mgr_find(tokdata, sSession->sessionh);
+    if (!sess) {
+        TRACE_ERROR("%s\n", ock_err(ERR_SESSION_HANDLE_INVALID));
+        rc = CKR_SESSION_HANDLE_INVALID;
+        goto done;
+    }
+
+    rc = session_mgr_cancel(tokdata, sess, flags);
+
+done:
+    TRACE_INFO("SC_SessionCancel: sess = %lu\n", sSession->sessionh);
```

Comment:
Yes thats intended. `sSession->sessionh` is the session handle , while `sess` is the pointer to the the session structure. The session handle is of more interest in the trace, than the pointer.

Reply:
`sSession->sessionh` is the value of the caller. But the caller does not know about the local variable `sess`. I would prefer to use the variable name in the message, which is known to the caller. Otherwise it might be a bit confusing.

Reply:
> sSession->sessionh is the value of the caller

Well, yes, and no. The sessions are managed in a 2 level btree setup. There is a session btree managed by the API layer, and there is a per token session btree managed by the token. `sSession->sessionh` is actually the session handle as in the token session btree. The external session handle is not known by the token (i.e. in new_host.c).
So tracing `sSession->sessionh` is only only somewhat useful, better would be to trace the external session handle, but in any case its better than tracing the session pointer.

Reply:
Ok, but log the "real" variable name could help a lot in debugging sessions.

Reply:
Its done that way in many other functions.... I would like to stick with how it is elsewhere (although I somewhat agree with you).

Reply:
ok. But please add to your ToDo list, that we have to re-design the tracing for all these cases.